server.js

```diff
@@ -2,33 +2,148 @@ import express from 'express'
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import places from "./data/dream-places.json"
+
+const fetch = require("node-fetch");
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+const shufflePlaces = () => {
+  return places.sort(() => Math.random() - 1)
+}
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  const { status, lucid, minSize, maxSize, keyword } = req.query
+  let filteredPlaces = places
+  if (keyword) {
+    fetch (`https://words.bighugelabs.com/api/2/5b150550ce000e98377ef0189eca303f/${keyword}/json`)
+    .then(res => res.json())
+    .then(similarWords => {
+      filteredPlaces = filteredPlaces.filter(place => {
+        let synonyms = []
+        if (similarWords.hasOwnProperty("noun")) {
+          synonyms = [...synonyms, ...similarWords.noun.syn]
+        }
+        if (similarWords.hasOwnProperty("adjective")) {
+          synonyms = [...synonyms, ...similarWords.adjective.syn]
+        }
+        if (similarWords.hasOwnProperty("verb")) {
+          synonyms = [...synonyms, ...similarWords.verb.syn]
+        }
+        return ( 
+          synonyms.some(word => place.keywords.includes(word))
+        )
+      })
+      if (filteredPlaces.length === 0) {
+        res.status(404).json({ error: `Couldn't find any locations related to keyword '${keyword}'.`})
+      } else {
+        res.status(200).json({ data: filteredPlaces })
+      }
+    })
+    .catch(err => console.log(err))
+  }
+  if (status) {
+    filteredPlaces = filteredPlaces.filter(place => place.status === status)
+  }
+  if (lucid) {
+    filteredPlaces = filteredPlaces.filter(place => place.governor === lucid.replace(/_|-|\./, " "))
+  }
+  if (minSize && maxSize) {
+    filteredPlaces = filteredPlaces.filter(place => 
+      parseInt(place.size.slice(0, place.size.length-1)) >= parseInt(minSize) 
+      && parseInt(place.size.slice(0, place.size.length-1)) <= parseInt(maxSize)
+    )
+  } else if (minSize) {
+    filteredPlaces = filteredPlaces.filter(place => 
+      parseInt(place.size.slice(0, place.size.length-1)) >= parseInt(minSize) 
+    )
+  } else if (maxSize) {
+    filteredPlaces = filteredPlaces.filter(place => 
+      parseInt(place.size.slice(0, place.size.length-1)) <= parseInt(maxSize)
+    )
+  }
+  if (keyword) {
+    //do nothing
+  } else {
+    if (filteredPlaces.length === 0) {
+      res.status(404).json({ error: `Couldn't find any locations related to keyword '${keyword}'.`})
```

Comment:
You repeating yourself a lot with these error actions. Maybe it is a great opportunity to create error handlers and call them here??